### PR TITLE
fix(kad): export `RoutingUpdate`

### LINKED ISSUE #3737: Kademlia::add_address return type is not exported (public)
## Description

The [`libp2p::kad::Kademlia::add_address`](https://docs.rs/libp2p/latest/libp2p/kad/struct.Kademlia.html#method.add_address) method returns a `RoutingUpdate` type which a developer can not inspect.

## Motivation

Knowing whether the address was successfully added or not is useful information that can be logged or else acted upon.

## Requirements

1. Export the [`RoutingUpdate`](https://github.com/libp2p/rust-libp2p/blob/7c85f92e31771b89f109f2e069eb8d0fea56bbdb/protocols/kad/src/behaviour.rs#L3176) type for use outside users.

## Are you planning to do it yourself in a pull request?

Yes.

### PR DESCRIPTION
**Pull request by b-zee**
## Description

The `RoutingUpdate` type that `add_address` returns was not made available publicly. This type is added in this PR by `pub use`.

Resolves #3737.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by thomaseizinger — APPROVED** [review]
Thanks!

**Comment by thomaseizinger** [comment]
Can you add a changelog entry and bump the version if there isn't yet an unreleased section in the changelog of `libp2p-kad`?

**Comment by b-zee** [comment]
I've changed the version from `0.43.1` to `0.43.2`. Not 100% if it should be `0.44.0`, but that would also involve updating dependants and their respective changelogs.

**Comment by dariusc93** [comment]
> I've changed the version from `0.43.1` to `0.43.2`. Not 100% if it should be `0.44.0`, but that would also involve updating dependants and their respective changelogs.

I believe the patch bump is fine to do

File: protocols/kad/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 0.43.2
```
Comment (thomaseizinger) [inline protocols/kad/CHANGELOG.md]:
```suggestion
## 0.43.2 - unreleased
```

**Review by thomaseizinger — DISMISSED** [review]
Thanks!

**Review by thomaseizinger — DISMISSED** [review]

File: protocols/kad/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## 0.43.2 - unreleased
+
+- Export pub enum `RoutingUpdate`. See [PR 3737].
+
+[PR 3737]: https://github.com/libp2p/rust-libp2p/pull/3737
```
Comment (thomaseizinger) [inline protocols/kad/CHANGELOG.md]:
```suggestion
- Export pub enum `RoutingUpdate`. See [PR 3739].

[PR 3739]: https://github.com/libp2p/rust-libp2p/pull/3739
```

Comment (b-zee) [reply protocols/kad/CHANGELOG.md]: Woops, good catch!

**Review by mxinden — DISMISSED** [review]
:pray: thanks for working on the patch!

**Pull request merged**